lib/ansible/modules/crypto/openssl_certificate.py

```diff
@@ -1871,11 +1871,11 @@ def _validate_csr_subject(self):
     def _validate_csr_extensions(self):
         cert_exts = self.cert.extensions
         csr_exts = self.csr.extensions
-        if len(cert_exts) != len(csr_exts):
+        if len(cert_exts) <= len(csr_exts):
```

Comment:
As mentioned on IRC, you really need this:
```suggestion
        if len(cert_exts) < len(csr_exts):
```